waiter/src/waiter/process_request.clj

```diff
@@ -235,9 +225,12 @@
         (log/error e "Unable to track content-length on request")))
     (when waiter-debug-enabled?
       (log/info "connecting to" instance-endpoint))
-    (make-http-request
-      http-client make-basic-auth-fn request-method instance-endpoint query-string headers body service-password
-      (handler/make-auth-user-map request) initial-socket-timeout-ms output-buffer-size proto-version)))
+    (-> instance
+        :protocol
+        (hu/retrieve-http-client http-clients)
```

Comment:
We choose the correct http client here to make either HTTP/1 or HTTP/2 requests.